### ISSUE TITLE
[css-counter-styles] How should relative-length values be interpreted when used in @counter-style descriptors?

### ISSUE THREAD

**Issue by weinig** [w3c/csswg-drafts#10877]
How should relative-length values be interpreted when used in @counter-style descriptors?

The grammars for @counter-style descriptors, for example, ['range'](https://drafts.csswg.org/css-counter-styles/#descdef-counter-style-range), allow the use of relative-length values due to the fact that `calc()` allows for them. 

For example, using 'range' again:

```css

@counter-style a {
  range: infinite calc(3 * sign(10em - 2px));
}
```

So the question is, what does 10em evaluate to? @font-palette-values contains this text:

> [Math functions](https://www.w3.org/TR/css-values-4/#math-function), such as [calc()](https://www.w3.org/TR/css-values-4/#funcdef-calc), and also [var()](https://www.w3.org/TR/css-variables-1/#funcdef-var), and [env()](https://drafts.csswg.org/css-env-1/#funcdef-env), are valid within descriptor values in a [@font-palette-values](https://www.w3.org/TR/css-fonts-4/#at-ruledef-font-palette-values) rule. They are evaluated within the context of the root element. Relative units are also evaluated within the context of the root element.

but I can't find any similar text for @counter-style.

**Comment by fantasai** [comment]
I don't think there's any real need for this, so if it's easier on implementations I think it's reasonable to disallow it.

**Comment by tabatkins** [comment]
I agree there's no particular use-case for it, but I do care that grammars are consistently interpreted. If we want to disallow math functions (but, presumably, allow other functions that could produce numbers, like custom `env()`), we should make a production with that restriction. If we want to allow only literals for some reason, we can use the `<number-token>` production.
